### PR TITLE
add color resolvable, and color constants from the client

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

File: src/client/ClientDataResolver.js [hunk 1]
```diff
@@ -317,6 +317,66 @@ class ClientDataResolver {
...
+   *   'DARK_PURPLE',
+   *   'DARK_GOLD',
+   *   'DARK_ORANGE',
+   *   'DARK_RED',
+   *   'DARK_GREY',
+   *   'LIGHT_GREY',
+   *   'DARK_NAVY',
+   * ]
+   * ```
+   * @typedef {String|number|Array} ColorResolvable
```
Comment (hydrabolt) [inline src/client/ClientDataResolver.js]: An array of what?

File: src/client/ClientDataResolver.js [hunk 2]
```diff
@@ -317,6 +317,66 @@ class ClientDataResolver {
...
+   * @param {ColorResolvable} color Color to resolve
+   * @returns {number} A color
+   */
+  static resolveColor(color) {
+    if (typeof color === 'string') {
+      if (color in Constants.Colors) {
+        color = Constants.Colors[color];
+      } else {
+        color = parseInt(color.replace('#', ''), 16);
+      }
```
Comment (hydrabolt) [inline src/client/ClientDataResolver.js]: 357-361 could just be simplified to `color = Constants.Colors[color] || parseInt(color.replace('#', ''), 16);`

File: src/client/rest/RESTMethods.js [hunk 3]
```diff
@@ -434,9 +434,7 @@ class RESTMethods {
     data.name = _data.name || role.name;
     data.position = typeof _data.position !== 'undefined' ? _data.position : role.position;
     data.color = _data.color || role.color;
-    if (typeof data.color === 'string' && data.color.startsWith('#')) {
-      data.color = parseInt(data.color.replace('#', ''), 16);
-    }
+    data.color = this.client.resolver.resolveColor(data.color);
```
Comment (hydrabolt) [inline src/client/rest/RESTMethods.js]: could just be `data.color = this.client.resolver.resolveColor(_data.color || role.color);` and remove 436

File: src/client/ClientDataResolver.js (hunk 1 above)
Comment (Tobsta) [inline src/client/ClientDataResolver.js]: @hydrabolt RGB values, like [255, 255, 255]

Comment (hydrabolt) [inline src/client/ClientDataResolver.js]: It should be clarified in the documentation

**Pull request merged**